### PR TITLE
Self Delegation

### PR DESCRIPTION
**Pull request by j-c17**
MVP , but may need refactor later.

### CODE REVIEW

File: Project/scripts/getWalletAccount.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ async function main(
       ? ethers.Wallet.fromMnemonic(mnemonic)
       : new ethers.Wallet(privateKey ?? EXPOSED_KEY);
   console.log(`Using address ${wallet.address}`);
-  const provider = ethers.providers.getDefaultProvider("rinkeby");
+  const provider = ethers.providers.getDefaultProvider("ropsten");
```
Comment (allen-hsu) [inline Project/scripts/getWalletAccount.ts]: I think can use @drsmog  situation

File: Project/scripts/deploymentAndMinting.ts [hunk 2]
```diff
@@ -3,7 +3,8 @@ import { ethers } from "ethers";
 import "dotenv/config";
```
Comment (allen-hsu) [inline Project/scripts/deploymentAndMinting.ts]: do we still need this, I think @drsmog https://github.com/Encode-Club-Solidity-Bootcamp-Group1/Week2-Tokenized-Votes/pull/2/files
split some task, I think we can use @drsmog version?

**Comment by j-c17** [comment]
@drsmog , I probably need your help on this one, since our tasks are very similar. 
The difference is that my one is self delegating. 
So the contract owner self delegating to himself and then vote. 
I compare my codes with your codes. It is pretty much the same. 
But I don't understand that after minting token and delegating to my self, the ballot contract returns me 
`Ballot Contract - Voting Power BEFORE Voting for Address 0x8C53D27499BF8d489b8a2a185147D786Dc907AD1 is 0.0` 

When I change the owner of the contract delegates to other address. The voting power is correct, but when that other address can't vote (`Error: cannot estimate gas; transaction may fail or may require manual gas limit`) 

When I run your codes, it work fine.  
Would you be OK to lend me a pair of eye to review my codes, what could be different than your code. 
Thanks a lot in advance.

**Comment by drsmog** [comment]
@j-c17 I've just reviewde PR checkout your branch and run operate.ts script and it works fine for me there are results

image

only changes i've made before run script was since u are using getSigner method which needs NETWORK to be defined in .evn i've just put ropsten there like this 
image

and another small thing i've just commented out signerone signertwo in script since in this example we just need one signer

image

So anyway it works for me correctly

**Pull request closed without merging**